gtfs/api/routes.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework import serializers, viewsets, filters
+from gtfs.models import Route, Stop
+from gtfs.api.stops import StopSerializer
+
+
+class RouteSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Route
+        fields = ("id", "name", "stops")
+
+    name = serializers.SerializerMethodField()
+    stops = serializers.SerializerMethodField()
+
+    def get_name(self, obj):
+        return obj.short_name
+
+    def get_stops(self, obj):
+        if "stops" not in self.context:
+            self.context["stops"] = Stop.objects.all()
+        stops = self.context["stops"].filter(feed=obj.feed)
+        return StopSerializer(stops, many=True).data
+
+
+class RoutesByStopIdFilterBackend(filters.BaseFilterBackend):
+    def filter_queryset(self, request, queryset, view):
+        query_params = request.query_params
+
+        if "stopId" not in query_params:
+            return queryset
+
+        stop_id = query_params["stopId"]
+        stop = Stop.objects.get(pk=stop_id)
+        queryset = queryset.filter(feed=stop.feed)
+        return queryset
```

Comment:
While implementing a custom `FilterBackend` like this works, you need to write a lot of functionality by hand which `django-filter` would provide. A better alternative would be to implement a [FilterSet](https://django-filter.readthedocs.io/en/master/ref/filterset.html). Also note the part about [DRF](https://django-filter.readthedocs.io/en/stable/guide/rest_framework.html#).